backend/util/notification.py

```diff
@@ -0,0 +1,131 @@
+        if value == notification_type:
+            return key
+    
+    return "00"
+
+
+class Notification(object):
+    """
+    Class to create notification.
+    """
+
+    def __init__(self, message, entity_key, notification_type, receiver_key):
+        """
+        Constructor of class Notification.
+
+        Keyword arguments:
+        message -- Message dictionary of notification.
+        entity_key -- entity key of type invite.
+        notification_type -- type of notification. 
+        receiver_key -- Urlsafe of the user who will receive the notification.
+        """
+        self.message = message
+        self.entity_key = entity_key
+        self.notification_type = self._get_notification_type(notification_type)
+        self.receiver_key = receiver_key
+        self.key = self._generate_key()
+    
+    def format_notification(self):
+        """
+        Method that returns a dictionary with the attributes needed to send the notification.
+        """
+        return {
+            'receiver_key': self.receiver_key,
+            'message': self.message,
+            'notification_type': self.notification_type,
+            'entity_key': self.entity_key
+        }
+    
+    def _generate_key(self):
+        """
+        Method that generates a unique key for the created notification. 
+        Using to generate the key the notification type id, the hash 
+        of the entity_key, the hash of the receiver_key, and the current 
+        date in seconds.
+        """
+        id = get_notification_id(self.notification_type)
+        entity_hash = hash(self.entity_key)
+        receiver_hash = hash(self.receiver_key)
+        timestamp = time.time()
+
+        key = id + '-' + str(entity_hash) + str(receiver_hash) + str(timestamp)
+        key = key.replace(".", "")
+        return key
+    
+    def _get_notification_type(self, notification_type):
+        """
+        Method to get the notification type according to the notifications_id dictionary. 
+        If the notification type already exists in the dictionary, it is returned, 
+        otherwise the default type 'ALL_NOTIFICATIONS' is returned.
+        """
+        return notification_type if notification_type in notification_id.values() else "ALL_NOTIFICATIONS"
+    
+    def send_notification(self):
+        """
+        Method to send notification.
+        """
+        send_message_notification(**self.format_notification())
+    
+    def __str__(self):
+        """
+        Method to generate string representation of notification object.
+        """
+        formated_notification = self.format_notification()
+        formated_notification['key'] = self.key
+
+        return self.__class__.__name__ + " " + str(formated_notification)
+    
+    def __repr__(self):
+        """
+        Method to generate string representation of notification object.
+        """
+        return str(self)
+
+
+class NotificationNIL(Notification):
```

Comment:
for what?

Reply:
For situations where the same worker may or may not send notifications on certain occasions.